spyder/plugins/ipythonconsole/widgets/client.py

```diff
@@ -504,7 +504,7 @@ def shutdown(self):
             self.shellwidget.spyder_kernel_comm.close()
             self.shellwidget.kernel_manager.shutdown_kernel(now=now)
         if self.shellwidget.kernel_client is not None:
-            self.shellwidget.kernel_client.shutdown()

```

Comment:
What's the difference between `shutdown` and `stop_channels`?

Reply:
`shutdown` stops the kernel, but is already called in `self.shellwidget.kernel_manager.shutdown_kernel(now=now)`. `stop_channels` stops the channels, including the heartbeat channel. If the heartbeat channel is not stopped but the kernel is down, the channel will notice and send `WARNING:traitlets:kernel died: 18.0`